http-auth/routes/routes_test.go

```diff
@@ -0,0 +1,166 @@
+package routes
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+)
+
+func TestMain(t *testing.T) {
+	t.Run("GET /", func(t *testing.T) {
+		request, _ := http.NewRequest(http.MethodGet, "/", nil)
+		rr := httptest.NewRecorder()
+
+		handler := http.HandlerFunc(IndexHandler)
+
+		handler.ServeHTTP(rr, request)
+
+		AssertStatusOK(t, rr)
+	})
+
+	t.Run("POST /", func(t *testing.T) {
+		bodyHTML := "<em>Hello Test</em>"
+		b := strings.NewReader(bodyHTML)
+		request, _ := http.NewRequest(http.MethodPost, "/", b)
+		request.Header.Add("Content-Type", "text/html")
+
+		rr := httptest.NewRecorder()
+
+		handler := http.HandlerFunc(IndexHandler)
+
+		handler.ServeHTTP(rr, request)
+
+		AssertStatusOK(t, rr)
+
+		AssertBodyContains(t, rr, bodyHTML)
```

Comment:
Should this test be testing _contains_ or _equals_? (I don't really have a good answer here, I think either is fine, I'm just curious what made you choose contains :))

Reply:
The reason I chose contains is to not be very strict when it comes to HTML, as it can be something that changes a lot. Not sure, for this simple test it can be strict equal but for more "dynamic" pages contains works better maybe...